compiler/record-minmax/include/RecordMinMax.h

```diff
@@ -23,14 +23,18 @@
 #include "MinMaxObserver.h"
 
 #include <memory>
+#include <thread>
 
 namespace record_minmax
 {
 
 class RecordMinMax
 {
 public:
-  explicit RecordMinMax() = default;
+  explicit RecordMinMax(int32_t num_threads) : _threads_size(num_threads)
```

Comment:
Our SE tool may complain implicit type conversion int32 -> uint32. Could you change dtype of num_threads to `uint32_t`?